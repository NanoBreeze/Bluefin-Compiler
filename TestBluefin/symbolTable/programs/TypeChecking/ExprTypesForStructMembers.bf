struct First {
	int a;
	bool b;
	int f() { return a; }
};

struct Second {
	First one;
	float two;
	float g(float x) { return x; }
};


int main() {
	First first;
	first.a + 5.2;
	while (first.b) {}
	
	Second second;
	second.one;
	second.one.b;
	second.one.a * second.two;
	second.one.f() - second.g(5);
}