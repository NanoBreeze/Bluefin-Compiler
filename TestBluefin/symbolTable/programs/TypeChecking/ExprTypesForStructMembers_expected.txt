a - t_int - t_int
x - t_float - t_float
first - t_First - t_First
first.a - t_int - t_float
5.2 - t_float - t_float
first.a+5.2 - t_float - t_float
first - t_First - t_First
first.b - t_bool - t_bool
second - t_Second - t_Second
second.one - t_First - t_First
second - t_Second - t_Second
second.one - t_First - t_First
second.one.b - t_bool - t_bool
second - t_Second - t_Second
second.one - t_First - t_First
second.one.a - t_int - t_float
second - t_Second - t_Second
second.two - t_float - t_float
second.one.a*second.two - t_float - t_float
second - t_Second - t_Second
second.one - t_First - t_First
second.one.f() - t_int - t_float
second - t_Second - t_Second
5 - t_int - t_float
second.g(5) - t_float - t_float
second.one.f()-second.g(5) - t_float - t_float
